apps/admin-server/src/pages/projects/[project]/widgets/documentmap/[id]/general.tsx

```diff
@@ -259,24 +263,47 @@ export default function DocumentGeneral(
 
         <FormField
           control={form.control}
-          name="url"
+          name="urlVisible"
           render={({ field }) => (
             <FormItem>
-              <FormLabel>Pagina met begleidende tekst</FormLabel>
-              <FormControl>
-                <Input
-                  placeholder="/path/to/page"
-                  defaultValue={field.value}
-                  onChange={(e) => {
-                    field.onChange(e);
-                    onFieldChange(field.name, e.target.value);
-                  }}
-                />
-              </FormControl>
+              <FormLabel>
+                Heeft een pagina met begleidende tekst?
+              </FormLabel>
+              <Switch.Root
+                className="block w-[50px] h-[25px] bg-stone-300 rounded-full relative focus:shadow-[0_0_0_2px] focus:shadow-black data-[state=checked]:bg-primary outline-none cursor-default"
+                onCheckedChange={(e: boolean) => {
+                  field.onChange(e);
+                  setUrlVisible(e)
+                }}
+                checked={field.value}>
+                <Switch.Thumb className="block w-[21px] h-[21px] bg-white rounded-full transition-transform duration-100 translate-x-0.5 will-change-transform data-[state=checked]:translate-x-[27px]" />
+              </Switch.Root>
+              <FormMessage />
             </FormItem>
           )}
         />
-
+        {urlVisible && (
+          <FormField
+            control={form.control}
+            name="url"
+            render={({ field }) => (
+              <FormItem>
+                <FormLabel>Link naar pagina</FormLabel>
```

Comment:
Omdat er ook nog een pagina komt voor de versie met reacties, is het denk ik handig om hier een duidelijkere titel voor te gebruiken.

```suggestion
                <FormLabel>Link naar pagina met begeleidende tekst</FormLabel>
```